WNPRC_Purchasing/src/client/actions.ts

```diff
@@ -38,9 +38,43 @@ export function getData(
     });
 }
 
+function getQCState(purchasingAdminModel: PurchaseAdminModel, requestOrder: RequestOrderModel, lineItems: LineItemModel[], qcStates: QCStateModel[])
+{
+    let qcState = purchasingAdminModel?.qcState || requestOrder.qcState;
+
+    // check for the final/complete state
+    let completeState = qcStates.filter((qcState) => {
+        return qcState.label === 'Order Complete';
+    });
+
+    // if 'order complete' state is set, return
+    if (completeState?.[0]?.rowId === qcState) {
+        return qcState;
+    }
+
+    // otherwise, identify if all the line items are received
+    let receivedCount = 0;
+    lineItems.forEach((lineItem: LineItemModel)=> {
+        if (lineItem.quantityReceived >= lineItem.quantity) {
+            ++receivedCount;
+        }
+    });
```

Comment:
You can use reduce for this.

`const receivedCount = lineItems.reduce((count, lineItem) => {
        if (lineItem.quantityReceived >= lineItem.quantity) {
            return count++;
        }
    }, 0);`

Reply:
this was not working as expected, i tried couple other variants as well. Left it as is